### PR TITLE
Add Remote Request question type

### PR DESCRIPTION
**Pull request by millerdev**
spec: https://docs.google.com/document/d/1x1xz2N7I_kTUjM4dFx31vt9EuzPrOWzSojnWD062XDs

The second commit fixes an unrelated test issue.

@orangejenny @emord cc @ctsims

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -1670,7 +1670,7 @@ define([
 
         this.$f.find('.fd-props-toolbar').html(this.getMugToolbar(mug));
         for (var i = 0; i < sections.length; i++) {
-            var section = sections[i];
+            var section = _.clone(sections[i]);
```
Comment (orangejenny) [inline src/core.js]: Why is this needed?

File: src/remoteRequest.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                },
+                parameters: {
+                    lstring: gettext('Parameters'),
+                    visibility: 'visible',
+                    presence: 'optional',
+                    widget: widgets.xPath,
+                    xpathType: "generic",
+                    serialize: mugs.serializeXPath,
+                    deserialize: mugs.deserializeXPath,
+                    help: gettext('A reference to a parameters node, "/data/request_params"'),
```
Comment (orangejenny) [inline src/remoteRequest.js]: This says it's a reference to a single node, but the input accepts any xpath expression. Can/should we do anything to limit what people put in this box?

Comment (orangejenny) [reply src/remoteRequest.js]: Also, should this say something like "Drag a question here" rather than talking about a node reference? I realize this is advanced functionality, but still.

File: src/remoteRequest.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            typeName: gettext('Remote Request'),
+            dataType: REMOTE_REQUEST_TYPE,
+            icon: 'fa fa-globe',
+            isDataOnly: true,
+            isTypeChangeable: false,
+            supportsDataNodeRole: true,
+            writeDataNodeXML: function (xmlWriter, mug) {
+                mug.form.vellum.data.remoteRequest.requestMugs.push(mug);
+            },
+            spec: {
```
Comment (orangejenny) [inline src/remoteRequest.js]: Can/should there be validation for this mug? Like requiring a URL to be provided?

**Comment by orangejenny** [comment]
Looks good overall, my comments are just questions.

**Comment by orangejenny** [comment]
(didn't mean to close this)

File: src/core.js (hunk 1 above)
Comment (millerdev) [reply src/core.js]: because `section` is mutated below on line [1676](https://github.com/dimagi/Vellum/pull/924/files#diff-7eb52b366866677666470e019283c8eaR1676)

File: src/remoteRequest.js (hunk 3 above)
Comment (millerdev) [reply src/remoteRequest.js]: Good idea.

File: src/remoteRequest.js (hunk 2 above)
Comment (millerdev) [reply src/remoteRequest.js]: Unfortunately we don't have a widget type that only accepts a single ref, so I opted to use `widgets.xPath` since it is a nicer experience than `widgets.text` for drag/drop, etc. I don't think this use case is important enough to add a new "ref" widget type.

Good suggestion on the wording change.
